File: be/app/Http/Controllers/MedicalUnitsController.php
<?php

namespace App\Http\Controllers;

use App\UnitType;
use Illuminate\Http\Request;
use App\MedicalUnit;
use App\Http\Requests;

class MedicalUnitsController extends Controller
{
    public function index()
    {
        $units = MedicalUnit::with(['unitType', 'medics', 'medics.rating'])->get();
        return $units;
    }

    public function getTypes(){
        return UnitType::all();
    }

    public function show(MedicalUnit $unit)
    {
        return $unit;
    }

    public function store(Request $request)
    {
        $unit = MedicalUnit::create($request->all());
        return response()->json($unit, 201);
    }

    public function update(Request $request, MedicalUnit $unit)
    {
        $unit->update($request->all());
        return response()->json($unit, 200);
    }

    public function delete(MedicalUnit $unit)
    {
        $unit->delete();
        return response()->json(null, 204);
    }
}

File: ruby_be/app/controllers/api/medical_units_controller.rb
module Api
  class MedicalUnitsController < ApplicationController
    before_action :set_unit, only: [:show, :update, :destroy]

    def index
      units = MedicalUnit.all
      render :json => units
    end

    def show
      unit = MedicalUnit.find(params[:id])
      render :json => unit
    end

    def create
      unit = MedicalUnit.new(medical_units_params)
      if unit.save
        render :json => unit
      else
        render json: {status: 403, data: unit.errors}
      end
    end

    def update
      if @unit.update(medical_units_params)
        render json: @unit
      else
        render json: {status: 403, error: @unit.errors}
      end
    end

    def destroy
      unit = MedicalUnit.find(params[:id])
      unit.destroy!
      render :json => unit
    end

    private

    def set_unit
      @unit = MedicalUnit.find(params[:id])
    end

    def medical_units_params
      params.permit(:name, :type, :location, :logo)
    end
  end
end

